website/docs/cloud-docs/integrations/kubernetes/k8s-ops-v2-migration.mdx

```diff
@@ -0,0 +1,104 @@
+
+## Prepare for the upgrade
+
+Before you migrate to Terraform Cloud Kubernetes Operator v2, you must first update v1 of the operator to the latest version, including the custom resource definitions.
+
+```shell-session
+$ helm upgrade --namespace ${RELEASE_NAMESPACE} ${RELEASE_NAME} hashicorp/terraform
+```
+
+Next, backup the workspace resources.
+
+```shell-session
+$ kubectl get workspace --all-namespaces -o yaml > backup_tfc_operator_v1.yaml
+```
+
+Finally, we suggest that you set the `applyMethod` of each workspace to `manual`. In version 2 of the Terraform Cloud Kubernetes Operator this is the default value, however explicitly configuring it with the value provides extra protection from unintended automatic applies during the upgrade process.
+
+```shell-session
+kubectl patch workspace <WORKSPACE_NAME> --type=merge --patch '{"spec": {"applyMethod": "manual"}}'
+```
+## Upgrade the operator
+
+Clone the Terraform Cloud Kubernetes Operator repository.
+
+```shell-session
+$ git clone https://github.com/hashicorp/terraform-cloud-operator.git
+```
+
+Change to the migration directory.
+
+```shell-session
+$ cd terraform-cloud-operator/docs/migration/crds
+```
+
+View the changes that **patch A** will apply to the workspace CRD.
+
+```shell-session
+$ kubectl diff -f workspaces_patch_a.yaml
+```
+
+Patch the workspace CRD with **patch A**. This patch adds `app.terraform.io/v1alpha2` support, but excludes `.status.runStatus` because it has a different format in `app.terraform.io/v1alpha1` and causes JSON unmarshalling issues.
+
+```shell-session
+$ kubectl patch crd workspaces.app.terraform.io --patch-file workspaces_patch_a.yaml
+```
+
+Follow the [installation guide](https://github.com/hashicorp/terraform-cloud-operator/blob/main/docs/installation.md) to install v2 of the operator.
+
+Next, create a Kubernetes secret to store the TFC API token following the [Usage Guide](https://github.com/hashicorp/terraform-cloud-operator/blob/main/docs/usage.md#prerequisites). The API token can be copied from the Kubernetes secret that you created for v1 of the operator. By default, this is named `terraformrc`. Use the `kubectl get secret` command to get the API token.
+
+```shell-session
+$ kubectl --namespace ${RELEASE_NAMESPACE} get secret terraformrc -o json | jq '.data.credentials' | tr -d '"' | base64 -d
+```
+
+If v2 of the operator pod fails to start, ensure that each `workspace` object has the `spec.token` value set. To do this, patch each `workspace` object using the `kubectl patch` command.
+
+```shell-session
+$ kubectl patch workspace <WORKSPACE_NAME> --type=merge --patch '{"spec": {"terraformVersion": "1.4.6", "token": {"secretKeyRef": {"key": "token", "name": "tfc-operator"}}}}'
+```
+
+Validate
+
+```shell-session
+$ kubectl logs -f <POD_NAME>
+```
```

Comment:
What output do we expect to see?

Reply:
/cc @arybolovlev

Reply:
The operator produces quite a lot of logs, and they are all either `INFO` or `ERROR`.

The reconciliation loop always starts with the following message:

```
INFO	Workspace Controller	{"workspace": {"name":"<WORKSPACE_NAME>","namespace":"<NAMESPACE>"}, "msg": "new reconciliation event"}
```

When everything goes well, we should see only `INFO` messages with the final message:

```
INFO	Workspace Controller	{"workspace": {"name":"<WORKSPACE_NAME>","namespace":"<NAMESPACE>"}, "msg": "successfully reconcilied workspace"}
```

In the case the operator encounters any issues, we would see an `ERROR` message, it has the following format:

```
ERROR	Workspace Controller	{"workspace": {"name":"<WORKSPACE_NAME>","namespace":"<NAMESPACE>"}, "msg": "<ERROR_MESSAGE>"}
```

Usually, the error message is quite informative and it shouldn't be a problem to understand it and fix it. Live example:

```
ERROR	Workspace Controller	{"workspace": {"name":"this","namespace":"default"}, "msg": "failed to get terraform cloud client", "error": "Secret \"tfc-operator\" not found"}
```

In short, we expect to see only `INFO` messages.